### PR TITLE
Don't include vrf pubkey in nodeID for rewards API

### LINKED ISSUE #2228: Some rewards are not sent by the api
Looks like not all rewards are sent via the api for a layer. See logs here: https://github.com/spacemeshos/explorer-frontend/issues/102 . This what causes the under-reporting of rewards in dash and explore.

### LINKED ISSUE #2366: unable to convert bytes to nodeid
I'm seeing this error from all nodes in the testnet (v0.1.28, tn127):

```
{
  "_index": "sm-2021.04.06",
  "_type": "_doc",
  "_id": "Y28MqXgBFRy6v6uwCc25",
  "_score": 1,
  "_source": {
    "@version": "1",
    "sm": {
      "errmsg": "Invalid input length, input too long",
      "N": "00000.defaultLogger",
      "L": "ERROR",
      "M": "unable to convert bytes to nodeid"
    },
    "name": "miner-39",
    "@timestamp": "2021-04-06T21:17:19.991Z",
    "stream": "stdout"
  },
  "fields": {
    "@timestamp": [
      "2021-04-06T21:17:19.991Z"
    ]
  }
}
```

Likely due to a bug in #2262 (or to an issue it uncovered).

### PR DESCRIPTION
**Pull request by sidsankhe**
## Motivation

Closes #2228
Closes #2366 

## Changes

There was an issue with the deserialization of the NodeID which resulted in no rewards being returned for specific miners. 
## TODO

- [x] Explain motivation or link existing issue(s)
- [ ] Test changes and document test plan
- [ ] Update documentation as needed

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

File: common/types/block_test.go [hunk 1]
```diff
@@ -44,25 +44,19 @@ func TestStringToNodeID(t *testing.T) {
...
 	r := require.New(t)
 	r.NoError(err, "Error converting string to NodeID")
 	r.Equal(nodeID1.Key, reversed.Key, "Node ID deserialization Key does not match")
-	r.Equal(nodeID1.VRFPublicKey, reversed.VRFPublicKey, "Node ID deserialization VRF Key does not match")
 
 	// Test too short
 	_, err = StringToNodeID(string(pubkey[:10]))
 	r.Error(err, "Expected error converting too-short string to NodeID")
 
-	// Test too long
```
Comment (lrettig) [inline common/types/block_test.go:61]: You should leave a "too long" test for two reasons:

- if we're really ignoring everything after the first 32 bytes, then you should test passing in > 32 bytes and making sure you only get the first 32 back
- we want a regression test, i.e., one that would've failed before this fix but passes now

File: mesh/mesh.go [hunk 2]
```diff
@@ -350,6 +350,7 @@ func (msh *Mesh) reInsertTxsToPool(validBlocks, invalidBlocks []*types.Block, l
 }
 
 func (msh *Mesh) applyState(l *types.Layer) {
+	msh.With().Info("applying state of the layer")
```
Comment (lrettig) [inline mesh/mesh.go]: Don't use `With()` without passing additional LoggableFields - or, better yet, add the layer here

File: mesh/mesh.go [hunk 3]
```diff
@@ -765,6 +766,7 @@ func (msh *Mesh) GetOrphanBlocksBefore(l types.LayerID) ([]types.BlockID, error)
 }
 
 func (msh *Mesh) accumulateRewards(l *types.Layer, params Config) {
+	msh.With().Info("Accumulating rewards")
```
Comment (lrettig) [inline mesh/mesh.go]: same comment as above

File: mesh/meshdb.go [hunk 4]
```diff
@@ -519,6 +519,7 @@ func (m *DB) writeTransactionRewards(l types.LayerID, accountBlockCount map[type
 	for account, smesherAccountEntry := range accountBlockCount {
 		for smesherString, cnt := range smesherAccountEntry {
 			smesherEntry, err := types.StringToNodeID(smesherString)
+			m.Info("writing reward to database")
```
Comment (lrettig) [inline mesh/meshdb.go]: add some more context to this log message to make it more useful: layer, reward amount, smesher, etc.

**Review by lrettig — CHANGES_REQUESTED** [review]

File: common/types/block_test.go (hunk 1 above)
Comment (sidsankhe) [reply common/types/block_test.go:61]: Makes sense, I added in the too-long tests for both the string and bytes deserialization

**Comment by lrettig** [comment]
I improved the logs here for the testnet branch and cherry-picked the change onto this branch as well

File: common/types/block_test.go [hunk 5]
```diff
@@ -57,6 +58,12 @@ func TestStringToNodeID(t *testing.T) {
 	_, err = StringToNodeID(string(pubkey[:10]))
 	r.Error(err, "Expected error converting too-short string to NodeID")
 
+	//Test too long : make sure that everything after the first 64 characters is ignored
+	longString := nodeID1.String() + string(pubkey2[:])
+	reversed, err = StringToNodeID(string(longString))
```
Comment (lrettig) [inline common/types/block_test.go]: isn't `longString` already a string? why do you need to use `string()` here?

File: common/types/block_test.go [hunk 6]
```diff
@@ -57,6 +58,12 @@ func TestStringToNodeID(t *testing.T) {
 	_, err = StringToNodeID(string(pubkey[:10]))
 	r.Error(err, "Expected error converting too-short string to NodeID")
 
+	//Test too long : make sure that everything after the first 64 characters is ignored
+	longString := nodeID1.String() + string(pubkey2[:])
```
Comment (lrettig) [inline common/types/block_test.go:62]: Insert after this line:
`r.Greater(len(longString), len(nodeID1.String()))`

(it's fairly obvious but it makes the test more readable and helps protect against future changes to the test - it's always good to be extra verbose in tests)

File: common/types/block_test.go [hunk 7]
```diff
@@ -79,4 +86,12 @@ func TestBytesToNodeID(t *testing.T) {
 	rand.Read(x[:])
 	_, err = BytesToNodeID(x[:])
 	r.Error(err, "Expected error converting too-short byte array to NodeID")
+
+	// Test too long
+	longSlice := append(bytes, x[:]...)
```
Comment (lrettig) [inline common/types/block_test.go:92]: as above, add a `t.Greater` test

**Review by lrettig — CHANGES_REQUESTED** [review]
Just some minor tweaks in the test and then I think this is good to go

**Comment by lrettig** [comment]
Please merge in develop and resolve merge conflicts. Probably also best to rebase this on #2411 before finalizing it.

**Review by lrettig — CHANGES_REQUESTED** [review]
Let's rebase on #2411 before finalizing

File: common/types/block_test.go [hunk 8]
```diff
@@ -42,27 +42,29 @@ func TestFields(t *testing.T) {
...
-	rand.Read(x[:])
-	_, err = StringToNodeID(string(x[:]))
-	r.Error(err, "Expected error converting too-long string to NodeID")
+	//Test too long : make sure that everything after the first 64 characters is ignored
+	longString := nodeID1.String() + string(pubkey2[:])
+	r.Greater(len(longString), len(nodeID1.String()))
+	reversed, err = StringToNodeID(longString)
+	r.NoError(err, "Error converting too-long string to NodeID")
+	r.Equal(nodeID1.Key, reversed.Key, "NodeID does not match for too-long string")
+	r.Equal([]byte{}, reversed.VRFPublicKey, "VRF Key is not empty for too-long string")
```
Comment (lrettig) [inline common/types/block_test.go:67]: "is" empty, not "is not" empty, right?

Comment (lrettig) [reply common/types/block_test.go:67]: also, is there a difference between `[]byte{}` and `[]byte(nil)` (which you use below)?

File: common/types/block_test.go [hunk 9]
```diff
@@ -88,8 +89,11 @@ func TestBytesToNodeID(t *testing.T) {
...
-	var y [65]byte
-	rand.Read(y[:])
-	_, err = BytesToNodeID(y[:])
-	r.Error(err, "Expected error converting too-long byte array to NodeID")
+	longSlice := append(bytes, x[:]...)
+	r.Greater(len(longSlice), len(bytes))
+	reversed, err = BytesToNodeID(longSlice)
+	r.NoError(err, "Error converting too-long byte array to NodeID")
+	r.Equal(nodeID1.Key, reversed.Key, "NodeID Key does not match for too-long")
+	r.Equal([]byte{}, reversed.VRFPublicKey, "VRF Key is not empty for too-long")
```
Comment (lrettig) [inline common/types/block_test.go:97]: is there a difference between `[]byte{}` and `[]byte(nil)` (which you use below)?

File: common/types/block.go [hunk 10]
```diff
@@ -134,15 +133,14 @@ func StringToNodeID(s string) (*NodeID, error) {
...
 	// portion of the string corresponding to the Edwards public key
+	// Note that we don't care when the input is too long, because we are only deserializing the first 32
+	// bytes into the Edwards public key. The VRFKey is not used by the deserializaiton at all, and thus
+	// it is returned as empty
 	pubKey := s[:64]
-	vrfKey := s[64:]
 	return &NodeID{
 		Key:          pubKey,
-		VRFPublicKey: []byte(vrfKey),
+		VRFPublicKey: []byte{},
```
Comment (lrettig) [inline common/types/block.go:143]: `[]byte{}` or `[]byte(nil)`?

File: common/types/block.go [hunk 11]
```diff
@@ -114,15 +115,13 @@ func BytesToNodeID(b []byte) (*NodeID, error) {
...
-
+	// Note that we don't care when the input is too long, because we are only deserializing the first 32
+	// bytes into the Edwards public key. The VRFKey is not used by the deserializaiton at all, and thus
+	// it is returned as empty
 	pubKey := b[0:32]
-	vrfKey := b[32:]
 	return &NodeID{
 		Key:          util.Bytes2Hex(pubKey),
-		VRFPublicKey: []byte(util.Bytes2Hex(vrfKey)),
+		VRFPublicKey: []byte{},
```
Comment (lrettig) [inline common/types/block.go:124]: `[]byte{}` or `[]byte(nil)`?

**Review by lrettig — APPROVED** [review]
Approving, but see comments about minor changes

**Comment by nkryuchkov** [comment]
@lrettig Should we fix your suggestions, merge develop into this branch and merge the PR?

**Comment by dshulyak** [comment]
We will use ecvrf with a single key at some point, no need to do anything about this change

**Pull request closed without merging**